### DISCUSSION TITLE
<title>Does github store the ip addresses that accessed a file in a repository

### DISCUSSION THREAD

**Discussion by kwd2** [community/community#125104] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hello,   I have a file stored in a repo.   This file was accessed by 5 unique individuals on April 1st, one of which was me.   Is it possible to retrieve the ip addresses that accessed the file?    I expect to have discovery power in 2-3 months and will be able to subpoena the information if it exists.  
thanks
Ken D.

**Comment by Janethq** [comment]
GitHub does not provide IP address information for users who access specific files in a repository. The IP addresses that GitHub provides are for their own servers and services, and are not associated with individual users. Additionally, GitHub's Terms of Service prohibit using their platform for any activity that would violate the privacy of their users. Therefore, it is not possible to retrieve the IP addresses of users who accessed a specific file in a GitHub repository.

**Comment by kwd2** [comment]
Hello Jane,
   thank you for this information.      Is it correct that If I were to get
a Court ordered subpoena, that the ip addresses that accessed the file are
retained and could be retrieved under appropriate circumstances?    There
were 4 unique users that accessed a file as part of a legal matter, and I
am going to ask the Court for discovery power to find the ip addresses that
accessed the file.
thanks
Ken D.

**Comment by ebndev** [comment]
Hi @kwd2,

This community is managed by GitHub employees, but it is not a support forum. The only place to go for answers to account-specific questions like this one is opening a ticket on our [Support page](https://support.github.com/contact). Please open a ticket with them.

This is a public forum, and the community is unable to assist with nor escalate your ticket, but we can assure you it is in the right place. It will be answered in the order it was received based on our team's resources and the amount of tickets ahead of yours.

As no other Community Discussions team members will be able to provide additional help with account related questions I am going to close this discussion. We appreciate your understanding!
